pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +255,93 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// TODO: change the name
+func (c *APIServiceController) filterAPIServices() []*apiregistrationv1.APIService {
+
+	oauthOperatorPrecondition := c.oauthOperatorAvailableAndVersionMatch()
+
+	filteredApiServices := []*apiregistrationv1.APIService{}
+
+	// the c.apiServices list is authoritative
+	// it can compete with oauth-server in case of errors but it could save us on downgrade (the current coao and an old cao)
+	for _, apiService := range c.apiServices {
+		if c.isAPIServicePotentiallyManagedByExternalServer(apiService) && oauthOperatorPrecondition && c.isAPIServiceAnnotatedByExternalServer(apiService) {
+			// TODO: log that this resource is being skipped on behalf of oauth-server
+			continue
+		}
+		filteredApiServices = append(filteredApiServices, apiService)
+	}
+
+	return filteredApiServices
+}
+
+func (c *APIServiceController) oauthOperatorAvailableAndVersionMatch() bool {
```

Comment:
this needs to be a function you pass in during construction.

Reply:
may I know why :) ?

Reply:
This is going to move to library-go so we can re-use it in openshift and oauth server operators